kvbc/include/kvbc_app_filter/kvbc_app_filter.h

```diff
@@ -57,54 +57,15 @@ struct KvbFilteredEventGroupUpdate {
   EventGroup event_group;
 };
 
-// We need to preserve state per client across calls to getNextEventGroupId method
-struct EventGroupClientState {
-  EventGroupClientState(const uint64_t pub_oldest,
-                        const uint64_t pub_newest,
-                        const uint64_t pvt_oldest,
-                        const uint64_t pvt_newest)
-      : public_offset(pub_oldest), private_offset(pvt_oldest) {
-    // Because of pruning, the oldest tag-specific event group might not be available anymore.
-    // Therefore, we have to calculate the first available event group ID for this (private) tag.
-    // Note: If *_oldest is 0 then all event groups were pruned or never existed.
-    curr_trid_event_group_id = 0;
-
-    if (pub_oldest == 0) {
-      if (pub_newest) {
-        // All public event groups were pruned
-        curr_trid_event_group_id += pub_newest;
-      }
-    } else {
-      curr_trid_event_group_id += pub_oldest;
-    }
-
-    if (pvt_oldest == 0) {
-      if (pvt_newest) {
-        // All private event groups were pruned
-        curr_trid_event_group_id += pvt_newest;
-      }
-    } else {
-      curr_trid_event_group_id += pvt_oldest;
-    }
-
-    // Note: curr_trid_event_group_id has to start at oldest - 1
-    if (pub_oldest && pvt_oldest) {
-      curr_trid_event_group_id -= 2;
-    } else if (pub_oldest || pvt_oldest) {
-      curr_trid_event_group_id -= 1;
-    }
-  }
-  // holds a batch of the global event group IDs ordered in the order in which they were generated
-  std::vector<uint64_t> event_group_id_batch{};
-  // keeps track of the global event group id read from event_group_id_batch
-  std::vector<uint64_t>::iterator it = event_group_id_batch.begin();
-  // the offset event group id for public/private event groups, everytime event_group_id_batch is populated with
-  // public/public event group ids, we need to save the offset that determines the next public/private event group id to
-  // be read from the tag table
-  uint64_t public_offset;
-  uint64_t private_offset;
-  // current tag-specific event_group_id
-  uint64_t curr_trid_event_group_id;
+struct FindGlobalEgIdResult {
+  // Global event group id matching the given external id
+  uint64_t global_id;
+  // Whether the event group is private to the client or public
+  bool is_private;
```

Comment:
Nit: I think we might consider calling this `is_public`, instead. Reason is we use `private` in other places to mean no client can see an event or a state key-value. When an event or state key-value can only seen by a set of clients, we call it `restricted`. Please correct me if I miss something.

Reply:
I follow your reasoning and it makes sense to me - will make the change. Thanks!